moto/iam/models.py

```diff
@@ -892,6 +898,10 @@ def create_role(
                     permissions_boundary
                 ),
             )
+        if [role for role in self.get_roles() if role.name == role_name]:
```

Comment:
`self.roles` is a `dict` indexed by role name. Please rely on that instead of getting a list of roles that need to be O(n) searched.

Reply:
_self.roles_ is indexed by role_id! Which is a randomly generated id, so that wouldn't work unfortunately.
The _get_role(role_name)_-function has the same conundrum - it has to search the entire list

Reply:
@mikegrima

Reply:
😄Well that's unfortunate.